### ISSUE TITLE
Truffle does not work with RPC providers that do not implement evm_snapshot

### ISSUE THREAD

**Issue by area** [trufflesuite/truffle#3437]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

Truffle no longer appears to work correctly with RPC providers that do not implement `evm_snapshot`

## Steps to Reproduce

```
npm init -y
npm i truffle ganache-cli
sed -i 's/snapshot/disablesnapshot/g' ./node_modules/ganache-cli/build/ganache-core.node.cli.js
npx truffle unbox metacoin 
```
The `sed` here disables the `evm_snapshot` endpoint for the instance of `ganache-cli` we're installing. You will need to say yes to the prompt from the `unbox` command.
```
sed -i '8,33{s|//||g}' truffle-config.js
sed -i 's/7545/8545/g' truffle-config.js
cp ./test/metacoin.js ./test/metacoin2.js
``` 
This sets up the config file correctly to connect to `ganache-cli`, and adds extra test blocks required to trigger the error.
In separate windows:
```
npx ganache-cli
npx truffle test
```

## Expected Behavior

The tests run correctly.

## Actual Results

The tests fail with the output

```
  1) Contract: MetaCoin
       "before all" hook: prepare suite for "should put 10000 MetaCoin in the first account":
     ReferenceError: config is not defined
      at TestRunner.initialize (node_modules/truffle/build/webpack:/packages/core/lib/testing/TestRunner.js:44:1)
      at Context.<anonymous> (node_modules/truffle/build/webpack:/packages/core/lib/testing/Test.js:315:1)
      at processImmediate (internal/timers.js:456:21)
```

## Environment

* Operating System: Ubuntu 20.04
* Ethereum client: ganache-cli / parity
* Truffle version (`truffle version`): `Truffle v5.1.48 (core: 5.1.48)`
* node version (`node --version`): `Node v12.16.3`
* npm version (`npm --version`):  `6.14.4`

## Additional investigation
 
I think this is because there is a global `config` floating around that [this](https://github.com/trufflesuite/truffle/blob/develop/packages/core/lib/testing/TestRunner.js#L44) line manages to use, that gets clobbered during the redeployment. The second redeployment required then fails.

This is supported by the fact that if you monkeypatch `truffle` thusly, the tests pass:
```
sed -i 's/this.config.resolver = new Resolver(config, true);/this.config.resolver = new Resolver(this.config, true);/g' ./node_modules/truffle/build/commands.bundled.js
```

Though there may be unintended consequences to this I haven't seen.

**Comment by gnidan** [comment]
Yikes! We'll have look into this. Thanks for reporting!

**Comment by eggplantzzz** [comment]
This is fixed in version 5.1.50.
